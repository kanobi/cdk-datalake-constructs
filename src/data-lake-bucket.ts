import * as iam from '@aws-cdk/aws-iam';
import * as s3 from '@aws-cdk/aws-s3';
import * as cdk from '@aws-cdk/core';

export interface DataLakeBucketProps {
  readonly bucketName: string;
  readonly dataCatalogAccountId: string;
  readonly logBucket: s3.Bucket;
  readonly crossAccount: boolean;
  readonly s3Properties: s3.BucketProps | undefined;
}

export class DataLakeBucket extends cdk.Construct {
  public readonly bucket: s3.Bucket;

  constructor(scope: cdk.Construct, id: string, props: DataLakeBucketProps) {
    super(scope, id);

    this.bucket = new s3.Bucket(this, 'datalake-bucket', {
      bucketName: props.bucketName,
      ...props.s3Properties,
      serverAccessLogsBucket: props.logBucket,
    });

    if (props.crossAccount) {
      this.bucket.addToResourcePolicy(
        new iam.PolicyStatement({
          resources: [
            this.bucket.arnForObjects('*'),
            this.bucket.bucketArn,
          ],
          actions: ['s3:List*', 's3:Get*'],
          principals: [
            new iam.ArnPrincipal(`arn:aws:iam::${props.dataCatalogAccountId}:root`),
          ],
        }),
      );
    }
  }
}
