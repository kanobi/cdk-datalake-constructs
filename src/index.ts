export * from './data-lake';
export * from './data-product';
export * from './pipeline';
export * from './data-lake-bucket';
export * from './utils';
export * from './data-streams/s3-delivery-stream';
export * from './data-streams/kinesis-stream';
export * from './data-streams/kinesis-ops';
export * from './etl/glue-crawler';
export * from './etl/glue-job';
export * from './etl/glue-job-ops';
export * from './etl/glue-table';
export * from './global/enums';
export * from './global/interfaces';
// export * from './etl/glue-notebook';
export * from './personas/data-lake-admin';
export * from './personas/data-lake-analyst';
export * from './personas/data-lake-creator';
// export * from './emr-cluster-sc'
// export * from './emr-studio'
