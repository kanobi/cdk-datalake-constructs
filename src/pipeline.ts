import * as events from '@aws-cdk/aws-events';
import * as glue from '@aws-cdk/aws-glue';
import * as lambda from '@aws-cdk/aws-lambda';
import * as s3 from '@aws-cdk/aws-s3';
import * as cdk from '@aws-cdk/core';

import { GlueJobType, GlueVersion, GlueWorkerType } from './etl/glue-job';

export interface JDBCProperties {
  readonly jdbc: string;
  readonly username: string;
  readonly password: string;
}

export interface StreamProperties {
  readonly streamName: string;
  readonly lambdaDataGenerator?: LambdaDataGeneratorProperties;
}

export interface S3Properties {
  readonly sourceBucketName: string;
  readonly sourceKeys: string[];
}

export enum DataSetLocation {
  RAW = 'raw',
  TRUSTED = 'trusted',
  REFINED = 'refined'
}

export interface TableProps {
  readonly tableName: string;
  readonly description: string;
  readonly partitionKeys: Array<glue.CfnTable.ColumnProperty | cdk.IResolvable> | cdk.IResolvable;
  readonly columns: Array<glue.CfnTable.ColumnProperty | cdk.IResolvable> | cdk.IResolvable;
  readonly parameters: {[param: string]: any};
  readonly serializationLibrary: string;
  readonly serdeParameters: {[param: string]: any};
  readonly inputFormat: string;
  readonly outputFormat: string;
  readonly catalogId: string;
}

export interface JobProperties {
  readonly name: string;
  readonly roleName?: string;
  readonly description?: string;
  readonly readAccessBuckets?: s3.IBucket[];
  readonly writeAccessBuckets?: s3.IBucket[];
  readonly glueVersion?: GlueVersion;
  readonly workerType: GlueWorkerType;
  readonly numberOfWorkers?: number;
  readonly maxCapacity?: number;
  readonly maxRetries?: number;
  readonly maxConcurrentRuns?: number;
  readonly jobScript: string;
  readonly jobArgs?: { [key: string]: string };
  readonly timeout?: number;
  readonly jobType: GlueJobType;
  readonly destinationLocation?: DataSetLocation;
}

export interface DataStreamProperties {
  readonly name: string;
  readonly destinationBucketName: string;
  readonly destinationPrefix: string;
  readonly dataCatalogOwner: DataCatalogOwner;
  readonly streamName: string;
  readonly lambdaDataGenerator: LambdaDataGeneratorProperties;
}

export interface LambdaDataGeneratorProperties {
  readonly code: lambda.Code;
  readonly handler: string;
  readonly timeout: cdk.Duration;
  readonly runtime: lambda.Runtime;
  readonly functionName: string;
  readonly schedule: events.Schedule;
  readonly ruleName: string;
}

export interface PipelineProperties {
  readonly type: DataPipelineType;
  readonly name: string;
  readonly destinationPrefix: string;
  readonly dataSetDropLocation: DataSetLocation;
  readonly s3Properties?: S3Properties;
  readonly streamProperties?: StreamProperties;
  readonly jdbcProperties?: JDBCProperties;
  readonly s3NotificationProps?: S3NotificationProperties;
  readonly table? : TableProps;
  readonly job?: JobProperties;
}

export interface DataCatalogOwner {
  readonly accountId: string;
}

export interface S3NotificationProperties {
  readonly event: s3.EventType;
  readonly prefix: string;
  readonly suffix: string;
}

export enum DataPipelineType {
  STREAM = 'stream',
  JDBC = 'jdbc',
  S3 = 's3'
}

export class Pipeline {
  public readonly type: DataPipelineType
  public readonly name: string
  public readonly destinationPrefix: string
  public readonly dataSetDropLocation: DataSetLocation
  public readonly s3Properties?: S3Properties
  public readonly streamProperties?: StreamProperties
  public readonly jdbcProperties?: JDBCProperties
  public readonly s3NotificationProps?: S3NotificationProperties
  public readonly table? : TableProps
  public readonly job?: JobProperties

  constructor(props: PipelineProperties) {
    this.type = props.type;
    this.name = props.name;
    this.dataSetDropLocation = props.dataSetDropLocation;
    this.destinationPrefix = props.destinationPrefix;
    this.jdbcProperties = props.jdbcProperties ? props.jdbcProperties : undefined;
    this.job = props.job ? props.job : undefined;
    this.s3NotificationProps = props.s3NotificationProps ? props.s3NotificationProps : undefined;
    this.s3Properties = props.s3Properties ? props.s3Properties : undefined;
    this.streamProperties = props.streamProperties ? props.streamProperties : undefined;
    this.table = props.table ? props.table : undefined;
  }
}