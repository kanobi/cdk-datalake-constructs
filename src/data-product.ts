import { Duration, RemovalPolicy } from 'aws-cdk-lib';
import * as s3 from 'aws-cdk-lib/aws-s3';
import { IDependable } from 'constructs';
import { Pipeline } from './pipeline';

export interface DataProductProperties {
  readonly accountId: string;
  readonly dataCatalogAccountId?: string;
  readonly databaseName: string;
  readonly pipelines: Pipeline[];
  readonly s3BucketProps?: s3.BucketProps;
}

export class DataProduct implements IDependable {
  readonly accountId: string;
  readonly dataCatalogAccountId?: string;
  readonly databaseName: string;
  readonly pipelines: Pipeline[];
  readonly s3BucketProps?: s3.BucketProps;

  constructor(props: DataProductProperties) {
    this.accountId = props.accountId;
    this.dataCatalogAccountId = props.dataCatalogAccountId;
    this.databaseName = props.databaseName;
    this.pipelines = props.pipelines;

    if (props.s3BucketProps) {
      this.s3BucketProps = props.s3BucketProps;
    } else {
      this.s3BucketProps = {
        removalPolicy: RemovalPolicy.DESTROY,
        autoDeleteObjects: true,
        lifecycleRules: [
          {
            expiration: Duration.days(30),
          },
        ],
      };
    }
  }
}