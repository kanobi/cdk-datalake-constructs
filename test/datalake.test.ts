import { App, Stack } from '@aws-cdk/core';
import { DataLake, Stage, Pipeline, DataProduct } from '../src';
import * as pipelines from '../test/pipelines';
import '@aws-cdk/assert/jest';

const region = 'us-east-1';
const stage = Stage.ALPHA;
const dataProductAccountId = '123456789012';

const pipes: Array<Pipeline> = [
  pipelines.ReviewsPipeline(),
  pipelines.IoTDataPipeline(dataProductAccountId, region, stage),
];

const taxiPipes: Array<Pipeline> = [
  pipelines.YellowPipeline(),
  pipelines.GreenPipeline(),
];

const dataProducts: Array<DataProduct> = [{
  pipelines: pipes,
  accountId: dataProductAccountId,
  databaseName: 'data-product',
},
{
  pipelines: taxiPipes,
  accountId: dataProductAccountId,
  databaseName: 'taxi-product',
}];

test('Check Resources', () => {
  const app = new App();
  const stack = new Stack(app, 'testStack');

  const datalake = new DataLake(stack, 'datalake', {
    accountId: dataProductAccountId,
    name: 'test-lake',
    region: region,
    stageName: stage,
    dataProducts: dataProducts,
    createDefaultDatabase: true,
  });

  expect(datalake.accountId).toMatch(dataProductAccountId);
  expect(datalake.stageName).toMatch(Stage.ALPHA);
  expect(datalake.region).toMatch(region);
  expect(Object.keys(datalake.dataSets).length).toEqual(4);
  expect(Object.keys(datalake.dataStreams).length).toEqual(1);

  expect(stack).toHaveResource('AWS::S3::Bucket');
});