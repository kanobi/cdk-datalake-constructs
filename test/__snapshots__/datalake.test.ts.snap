// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cdk-nag AwsSolutions Pack Should match snapshot 1`] = `
Object {
  "Outputs": Object {
    "datalakeDataLakeAdminRole3EA5FF19": Object {
      "Value": Object {
        "Ref": "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
      },
    },
    "datalakeDataLakeAthenaWorkgroup48F354CE": Object {
      "Value": "test-lake-workgroup-alpha",
    },
    "datalakeDataLakeLogBucket7E4B6A14": Object {
      "Value": Object {
        "Ref": "datalakedatalakelogbucket0A814944",
      },
    },
    "datalaketestlakedatalakecreatorroleDataLakeDatabaseCreatorRoleC012854C": Object {
      "Value": Object {
        "Ref": "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9",
      },
    },
  },
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": Object {
      "DependsOn": Array [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "483ae06ed27ef8ca76e011264d772420593a6cfe8544759c306ef3b98c9e25be.zip",
        },
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Lambda function for auto-deleting objects in ",
              Object {
                "Ref": "datalakedatalakelogbucket0A814944",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "datalakedatalakelogbucket0A814944": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AccessControl": "LogDeliveryWrite",
        "BucketName": "test-lake-log-bucket-alpha",
        "LifecycleConfiguration": Object {
          "Rules": Array [
            Object {
              "ExpirationInDays": 7,
              "Status": "Enabled",
            },
          ],
        },
        "Tags": Array [
          Object {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "datalakedatalakelogbucketAutoDeleteObjectsCustomResource3DCF9F10": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "datalakedatalakelogbucketPolicyD1691576",
      ],
      "Properties": Object {
        "BucketName": Object {
          "Ref": "datalakedatalakelogbucket0A814944",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "datalakedatalakelogbucketPolicyD1691576": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "datalakedatalakelogbucket0A814944",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "datalakedatalakelogbucket0A814944",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "datalakedatalakelogbucket0A814944",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "datalakelfdatalakeroleadminsettings2C1ACF95": Object {
      "DependsOn": Array [
        "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
      ],
      "Properties": Object {
        "Admins": Array [
          Object {
            "DataLakePrincipalIdentifier": Object {
              "Fn::GetAtt": Array [
                "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
                "Arn",
              ],
            },
          },
        ],
      },
      "Type": "AWS::LakeFormation::DataLakeSettings",
    },
    "datalakereviewsproductdatabaseC2ED5A77": Object {
      "Properties": Object {
        "CatalogId": Object {
          "Ref": "AWS::AccountId",
        },
        "DatabaseInput": Object {
          "Name": "reviews-product",
        },
      },
      "Type": "AWS::Glue::Database",
    },
    "datalakereviewsproductlfdbcreatorpermission342BCB9F": Object {
      "DependsOn": Array [
        "datalakereviewsproductdatabaseC2ED5A77",
      ],
      "Properties": Object {
        "DataLakePrincipal": Object {
          "DataLakePrincipalIdentifier": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9",
              "Arn",
            ],
          },
        },
        "Permissions": Array [
          "ALTER",
          "CREATE_TABLE",
          "DROP",
        ],
        "Resource": Object {
          "DatabaseResource": Object {
            "Name": "reviews-product",
          },
        },
      },
      "Type": "AWS::LakeFormation::Permissions",
    },
    "datalaketaxiproductdatabase4B8A23BF": Object {
      "Properties": Object {
        "CatalogId": Object {
          "Ref": "AWS::AccountId",
        },
        "DatabaseInput": Object {
          "Name": "taxi-product",
        },
      },
      "Type": "AWS::Glue::Database",
    },
    "datalaketaxiproductlfdbcreatorpermission5C31145A": Object {
      "DependsOn": Array [
        "datalaketaxiproductdatabase4B8A23BF",
      ],
      "Properties": Object {
        "DataLakePrincipal": Object {
          "DataLakePrincipalIdentifier": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9",
              "Arn",
            ],
          },
        },
        "Permissions": Array [
          "ALTER",
          "CREATE_TABLE",
          "DROP",
        ],
        "Resource": Object {
          "DatabaseResource": Object {
            "Name": "taxi-product",
          },
        },
      },
      "Type": "AWS::LakeFormation::Permissions",
    },
    "datalaketestlakedatalakeadminroledatalakeadministratorTBACC6D7714B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "lakeformation:AddLFTagsToResource",
                "lakeformation:RemoveLFTagsFromResource",
                "lakeformation:GetResourceLFTags",
                "lakeformation:ListLFTags",
                "lakeformation:CreateLFTag",
                "lakeformation:GetLFTag",
                "lakeformation:UpdateLFTag",
                "lakeformation:DeleteLFTag",
                "lakeformation:SearchTablesByLFTags",
                "lakeformation:SearchDatabasesByLFTags",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "datalaketestlakedatalakeadminroledatalakeadministratorTBACC6D7714B",
        "Roles": Array [
          Object {
            "Ref": "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "datalaketestlakedatalakeadminroledatalakeadministratorbasic42E4DDDE": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "iam:CreateServiceLinkedRole",
              "Condition": Object {
                "StringEquals": Object {
                  "iam:AWSServiceName": "lakeformation.amazonaws.com",
                },
              },
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "iam:PutRolePolicy",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:iam::",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":role/aws-service-role/lakeformation.amazonaws.com/AWSServiceRoleForLakeFormationDataAccess",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "datalaketestlakedatalakeadminroledatalakeadministratorbasic42E4DDDE",
        "Roles": Array [
          Object {
            "Ref": "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "datalaketestlakedatalakeadminroledatalakeadministratorcrossaccountCA5F9CEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ram:AcceptResourceShareInvitation",
                "ram:RejectResourceShareInvitation",
                "ec2:DescribeAvailabilityZones",
                "ram:EnableSharingWithAwsOrganization",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "datalaketestlakedatalakeadminroledatalakeadministratorcrossaccountCA5F9CEB",
        "Roles": Array [
          Object {
            "Ref": "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "datalaketestlakedatalakeadminroledatalakeadministratorlambdawriteCWlogs84641127": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "AllowLogging",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "datalaketestlakedatalakeadminroledatalakeadministratorlambdawriteCWlogs84641127",
        "Roles": Array [
          Object {
            "Ref": "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lakeformation.amazonaws.com",
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "sagemaker.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AWSLakeFormationDataAdmin",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AWSGlueConsoleFullAccess",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/CloudWatchLogsReadOnlyAccess",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AWSLakeFormationCrossAccountManager",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonAthenaFullAccess",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonSageMakerFullAccess",
              ],
            ],
          },
        ],
        "RoleName": "test-lake-datalake-admin-alpha",
      },
      "Type": "AWS::IAM::Role",
    },
    "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaDefaultPolicy03434DE7": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lakeformation:GetDataAccess",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaDefaultPolicy03434DE7",
        "Roles": Array [
          Object {
            "Ref": "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "glue.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSGlueServiceRole",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AWSLakeFormationDataAdmin",
              ],
            ],
          },
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonS3FullAccess",
              ],
            ],
          },
        ],
        "Path": "/service-role/",
        "RoleName": "test-lake-datalake-creator-alpha",
      },
      "Type": "AWS::IAM::Role",
    },
    "datalakeworkgroup682714D0": Object {
      "Properties": Object {
        "Description": "Default Data Lake Workgroup",
        "Name": "test-lake-workgroup-alpha",
        "RecursiveDeleteOption": true,
        "State": "ENABLED",
        "WorkGroupConfiguration": Object {
          "EnforceWorkGroupConfiguration": true,
          "EngineVersion": Object {
            "EffectiveEngineVersion": "Athena engine version 2",
            "SelectedEngineVersion": "Athena engine version 2",
          },
          "ResultConfiguration": Object {
            "OutputLocation": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "s3://",
                  Object {
                    "Ref": "datalakedatalakelogbucket0A814944",
                  },
                  "/results/",
                ],
              ],
            },
          },
        },
      },
      "Type": "AWS::Athena::WorkGroup",
    },
    "reviewsdatasetstackNestedStackreviewsdatasetstackNestedStackResource3B660F85": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "Parameters": Object {
          "referencetotestdatalakedatalakelogbucket7EE4523BRef": Object {
            "Ref": "datalakedatalakelogbucket0A814944",
          },
          "referencetotestdatalaketestlakedatalakeadminroledatalakeadministratorrole4029157BArn": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
              "Arn",
            ],
          },
          "referencetotestdatalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralpha2AEEF09DArn": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9",
              "Arn",
            ],
          },
        },
        "TemplateURL": Object {
          "Fn::Join": Array [
            "",
            Array [
              "https://s3.",
              Object {
                "Ref": "AWS::Region",
              },
              ".",
              Object {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              Object {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/16329d110accc6fc933f0f598753c61fbe14b9d76cf40fe11bd91cb045acf693.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "taxigreendatasetstackNestedStacktaxigreendatasetstackNestedStackResource56C75800": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "Parameters": Object {
          "referencetotestdatalakedatalakelogbucket7EE4523BRef": Object {
            "Ref": "datalakedatalakelogbucket0A814944",
          },
          "referencetotestdatalaketestlakedatalakeadminroledatalakeadministratorrole4029157BArn": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
              "Arn",
            ],
          },
          "referencetotestdatalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralpha2AEEF09DArn": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9",
              "Arn",
            ],
          },
        },
        "TemplateURL": Object {
          "Fn::Join": Array [
            "",
            Array [
              "https://s3.",
              Object {
                "Ref": "AWS::Region",
              },
              ".",
              Object {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              Object {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/3bd9052ea75414d15d886bac69f40cba9ebb7a00dd4d38d64351ce8c5885e842.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "taxiyellowdatasetstackNestedStacktaxiyellowdatasetstackNestedStackResource56DD6B4B": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "Parameters": Object {
          "referencetotestdatalakedatalakelogbucket7EE4523BRef": Object {
            "Ref": "datalakedatalakelogbucket0A814944",
          },
          "referencetotestdatalaketestlakedatalakeadminroledatalakeadministratorrole4029157BArn": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakeadminroledatalakeadministratorrole0CE6C3E6",
              "Arn",
            ],
          },
          "referencetotestdatalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralpha2AEEF09DArn": Object {
            "Fn::GetAtt": Array [
              "datalaketestlakedatalakecreatorroleAWSDBCreatorServiceRoletestlakedatalakecreatoralphaF5D800D9",
              "Arn",
            ],
          },
        },
        "TemplateURL": Object {
          "Fn::Join": Array [
            "",
            Array [
              "https://s3.",
              Object {
                "Ref": "AWS::Region",
              },
              ".",
              Object {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              Object {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/47773d6a5cd15609edd23b3c773b54bcfaa6c607b5008692a7450eabf3ae6a5b.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
